src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java

```diff
@@ -172,9 +174,17 @@ static String getNextFieldPlaceholder(Map<String, String> expressionAttributeNam
         if (start == -1) {
             return Optional.empty();
         }
-        int end = conditionExpression.indexOf(" ", start + toFind.length()); // -1
-        return Optional.of(conditionExpression.substring(start + toFind.length(),
-            end == -1 ? conditionExpression.length() : end)); // ":value" // TODO add support for non-EQ operators
+
```

Comment:
why not replace this whole method with

        if (conditionExpression == null) {
            return Optional.empty();
        }

        final Pattern p = Pattern.compile(keyFieldName + " = ([^ ,]*)");
        final Matcher m = p.matcher(conditionExpression);
        return m.find() ? Optional.of(m.group(1)) : Optional.empty();

Reply:
alrighty then, fixed ... 541fcfbe15405c70d04d194837df5f7045da97fa